include/swift/Reflection/TypeRef.h

```diff
@@ -137,6 +139,49 @@ class TypeRefBuilder;
 using DepthAndIndex = std::pair<unsigned, unsigned>;
 using GenericArgumentMap = llvm::DenseMap<DepthAndIndex, const TypeRef *>;
 
+/// FIXME: Implement me!
+struct TypeRefLayoutConstraint {
+  friend llvm::hash_code hash_value(const TypeRefLayoutConstraint &layout) {
+    return llvm::hash_value(0);
+  }
+  operator bool() const { return true; }
+  bool operator==(TypeRefLayoutConstraint rhs) const {
+    return true;
+  }
+};
+
+#if LLVM_PTR_SIZE == 4
+/// Unfortunately the alignment of TypeRef is too large to squeeze in 3 extra
```

Comment:
Too small, not too large you mean? You could add an alignas declaration to it. We do this for Type and various other types in the rest of the compiler, too.

Reply:
I increased it from `alignas (void*)` to `alignas(8)`. Much cleaner.